infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -424,30 +587,35 @@ public void onNext(CorfuStreamEntries results) {
                     clusterManager.getClusterManagerCallback()
                             .applyNewTopologyConfig(clusterManager.generateConfigWithInvalid());
                 } else if (entry.getKey().equals(OP_RESUME)) {
+                    clusterManager.createSingleSourceSinkTopologyBuckets();
                     clusterManager.getClusterManagerCallback()
-                            .applyNewTopologyConfig(clusterManager.generateDefaultValidConfig());
+                            .applyNewTopologyConfig(clusterManager.topologyConfig);
                 } else if (entry.getKey().equals(OP_ENFORCE_SNAPSHOT_FULL_SYNC)) {
                     try {
+                        ClusterDescriptor localCluster = clusterManager.findLocalCluster();
+                        if (localCluster.getRole().equals(ClusterRole.SINK)) {
```

Comment:
check to be modified?

Reply:
No, this is the implementation of the clusterManager adapter that is used for testing.
So for testing purpose and to simulate various kinds of topology, I am continuing to use the role.

We can sink up offline if you have more questions about this.